misc/testerina/modules/testerina-core/src/main/ballerina/mock.bal

```diff
@@ -141,8 +160,122 @@ public class MemberFunctionStub {
     # + returnValue - Value or error to return
     public isolated function thenReturn(any|error returnValue) {
         if (self.functionName == "") {
-             error err = error("function to mock is not specified.");
-             panic err;
+            error err = error("function to mock is not specified.");
+            panic err;
+        }
+        self.returnValue = returnValue;
+        error? thenReturnExtResult = thenReturnExt(self);
+        if (thenReturnExtResult is error) {
+            panic thenReturnExtResult;
+        }
+    }
+
+    # Sets the values to be returned when the function is called repeatedly.
+    #
+    # + returnValues - Value or error to return
+    public isolated function thenReturnSequence(any|error... returnValues) {
+        if (self.functionName == "") {
+            error err = error("function to mock is not specified.");
+            panic err;
+        }
+        if (self.args.length() != 0) {
+            error err = error("'withArguments' function cannot be specified with a return sequence");
```

Comment:
Can we make this a compile-time error?

Reply:
Looks like this is the implementation in other places also. Let's address it in another Pr as an improvement.

Reply:
This improvement can be tracked with https://github.com/ballerina-platform/ballerina-lang/issues/42359